lib/backup_restore/backup_store.rb

```diff
@@ -0,0 +1,76 @@
+module BackupRestore
+  # @abstract
+  class BackupStore
+    class BackupFileExists < RuntimeError; end
+    class StorageError < RuntimeError; end
+
+    # @return [BackupStore]
+    def self.create(opts = {})
+      case SiteSetting.backup_location
+      when 'local'
+        require_dependency "backup_restore/local_backup_store"
+        BackupRestore::LocalBackupStore.new(opts)
+      when 's3'
+        require_dependency "backup_restore/s3_backup_store"
+        BackupRestore::S3BackupStore.new(opts)
+      end
+    end
+
+    # @return [Array<BackupFile>]
+    def files
+      unsorted_files
```

Comment:
Perhaps we should memoize this?

Reply:
I think it's fine the way it is. Currently `files` and `latest_file` isn't used more than once per `BackupStore` instance. Memoizing would just make it more complex, since we would need to invalidate or update it when files are added or deleted.